src/main/java/hlf/java/rest/client/config/KafkaConsumerErrorHandler.java

```diff
@@ -0,0 +1,114 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.TopicPartition;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.cloud.context.config.annotation.RefreshScope;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.listener.CommonErrorHandler;
+import org.springframework.kafka.listener.ConsumerAwareRecordRecoverer;
+import org.springframework.kafka.listener.DeadLetterPublishingRecoverer;
+import org.springframework.kafka.listener.DefaultErrorHandler;
+import org.springframework.util.backoff.FixedBackOff;
+
+@Configuration
+@ConditionalOnProperty("kafka.integration-points[0].brokerHost")
+@RefreshScope
+@Slf4j
+public class KafkaConsumerErrorHandler {
+
+  private static final long RETRY_INTERVAL_IN_MS = 1000L;
+  private static final int MAX_RETRY_ATTEMPTS = 5;
+  private static final List<Class<? extends Exception>> connectorRetryableExceptions =
+      Arrays.asList(RetryableServiceException.class, FabricTransactionException.class);
+
+  @Autowired private KafkaProperties kafkaProperties;
+
+  @Autowired private KafkaProducerConfig kafkaProducerConfig;
+
+  @Bean
+  public CommonErrorHandler topicTransactionErrorHandler() {
+
+    ConsumerAwareRecordRecoverer deadLetterPublishingRecoverer = null;
+
+    /**
+     * Check if the runtime config has a valid Failed Message Listener. If a dedicated Failed
+     * message listener is available, then the dead letter configuration would push the failed
+     * message to the exclusive dead letter topic. If a dedicated failed message listener is not
+     * configured, check if the event publisher topic can be utilised for publishing the failed
+     * message.
+     */
+    if (Objects.nonNull(kafkaProperties.getFailedMessageListener())
+        && StringUtils.isNotBlank(kafkaProperties.getFailedMessageListener().getTopic())) {
+
+      KafkaTemplate<String, String> deadLetterPublisherTemplate =
+          new KafkaTemplate<>(
+              kafkaProducerConfig.eventProducerFactory(kafkaProperties.getFailedMessageListener()));
+
+      // Set the exclusive dead letter topic
+      deadLetterPublisherTemplate.setDefaultTopic(
+          kafkaProperties.getFailedMessageListener().getTopic());
+      deadLetterPublishingRecoverer =
+          new DeadLetterPublishingRecoverer(
+              deadLetterPublisherTemplate,
+              (cr, e) ->
+                  new TopicPartition(
+                      kafkaProperties.getFailedMessageListener().getTopic(), cr.partition()));
```

Comment:
[nitpick] a method can be extracted for these lines as only listener is different and rest is same in both scenarios

Reply:
Makes sense. Done.